spark/v3.1/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestAlterTablePartitionFields.java

```diff
@@ -358,21 +384,34 @@ public void testReplaceNamedPartition() {
 
     sql("ALTER TABLE %s REPLACE PARTITION FIELD day_col WITH hours(ts)", tableName);
     table.refresh();
-    expected =
-        PartitionSpec.builderFor(table.schema())
-            .withSpecId(2)
-            .alwaysNull("ts", "day_col")
-            .hour("ts")
-            .build();
+    if (formatVersion == 1) {
+      expected =
+          PartitionSpec.builderFor(table.schema())
+              .withSpecId(2)
+              .alwaysNull("ts", "day_col")
+              .hour("ts")
+              .build();
+    } else {
+      expected =
+          PartitionSpecParser.fromJson(
+              table.schema(),
```

Comment:
Not a blocker but I think we should expose whatever we need to do this programmatically to testing so we don't have to parse jsons.

Reply:
Filed #8434.